### PR TITLE
feat(scanner): support github in scanner

### LINKED ISSUE #2450: [Feature] Support Github actions in scanner
### Description

Fossology scanner [does not support](https://github.com/fossology/fossology/blob/4.2.1/utils/automation/fossologyscanner.py#L41-L47) Github actions, only Gitlab and TravisCI.

#### How to reproduce

Need: ability to use `scanner` in Github actions.

### PR DESCRIPTION
**Pull request by imishchuk-carbon**
## Description

Add support of Github actions in `scanner`

### Changes

* Add logic to execute `scanner` in Github actions

## How to test

```
- name: Fossology scanner
  run: |
    docker run --name "fossologyscanner" -w "/opt/repo" -v ${PWD}:/opt/repo \
      -e GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }} \
      -e GITHUB_PULL_REQUEST=${{ github.event.number }} \
      fossology/fossology:scanner "/bin/fossologyscanner" nomos ojo
```

Note, `GITHUB_REPOSITORY` and `GITHUB_ACTIONS` are provided by Github action itself, so we do not need to specify it explicitly.
   

This closes #2450 

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2451"></a>

### CODE REVIEW

**Review by mhorbul — APPROVED** [review]
LGTM

File: utils/automation/fossologyscanner.py [hunk 1]
```diff
@@ -48,6 +54,11 @@ def get_ci_name():
     RUNNING_ON = 'TRAVIS'
     TRAVIS_REPO_SLUG = os.environ['TRAVIS_REPO_SLUG']
     TRAVIS_PULL_REQUEST = os.environ['TRAVIS_PULL_REQUEST']
+  elif 'GITHUB_ACTIONS' in os.environ and os.environ['GITHUB_ACTIONS'] == 'true':
+    RUNNING_ON = 'GITHUB'
+    GITHUB_TOKEN = os.environ['GITHUB_TOKEN']
+    GITHUB_REPOSITORY = os.environ['GITHUB_REPOSITORY']
+    GITHUB_PULL_REQUEST = os.environ['GITHUB_PULL_REQUEST']
```
Comment (GMishx) [inline utils/automation/fossologyscanner.py:61]: Unable to find documentation for `GITHUB_PULL_REQUEST`. Can you please share the same?

**Review by GMishx — COMMENTED** [review]
Thank you for your contribution @imishchuk-carbon . Can you please help with following comment?

Also, can you please add a sample GitHub action to the folder [utils/automation](https://github.com/fossology/fossology/tree/master/utils/automation) like the GitLab and Travis sample files?

File: utils/automation/fossologyscanner.py (hunk 1 above)
Comment (GMishx) [reply utils/automation/fossologyscanner.py:61]: Got it from the description. It is coming from `-e GITHUB_PULL_REQUEST=${{ github.event.number }}`

**Comment by imishchuk-carbon** [comment]
caught a bug fixed in this [commit](https://github.com/fossology/fossology/pull/2451/commits/0e20f4f705ffd0324c5cac9ae519787b442617cd). Not sure if this is applicable to Gitlab or Travis, though.

For example following change is present in PR

```
[
  {
    "sha": "fabed087f92842ef3ce2de181bd31c43434767ba",
    "filename": "some-folder/some-file.tsx",
    "status": "renamed",
    "additions": 0,
    "deletions": 0,
    "changes": 0,
    "blob_url": "https://github.com/some-org/some-repo/blob/3311cc4d9f9b2a42cc75803747cacad33fdf0faf/fsome-folder/some-file.tsx",
    "raw_url": "https://github.com/some-org/some-repo/blob/3311cc4d9f9b2a42cc75803747cacad33fdf0faf/fsome-folder/some-file.tsx",
    "previous_filename": "old-folder/some-file.tsx"
  },
```

This change is missing `change_key = "patch"` which leads to following exception

```
Traceback (most recent call last):
  File "/bin/ch-fossologyscanner", line 594, in <module>
    sys.exit(main(sys.argv))
  File "/bin/ch-fossologyscanner", line 541, in main
    cli_options.diff_dir = repo_setup.get_diff_dir()
  File "/bin/ch-fossologyscanner", line 208, in get_diff_dir
    print(re.sub(remove_diff_regex, r"\2", change[change_key]),
KeyError: 'patch'
```

**Comment by GMishx** [comment]
Hey @imishchuk-carbon , can you please do the following changes so we can proceed with merge?

1. Provide a sample yaml file for GitHub Action under `utils/automation/.github-workflow.yml`
2. Please rebase your branch with upstream master and squash the commits. We are currently seeing 14 commits on this PR. We follow atomic commits, so there should be only 1 commit for a single change.

**Comment by imishchuk-carbon** [comment]
Hey @GMishx 
Requested items have been done.

**Review by GMishx — APPROVED** [review]
Changes looks good.

File: utils/automation/.github-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Check diff for license with Fossology
```
Comment (GMishx) [inline utils/automation/.github-workflow.yml:1]: Looks like you are missing licensing and copyright information for this file:
https://github.com/fossology/fossology/actions/runs/5430470889/jobs/9878340162?pr=2451#step:4:7

You can refer to other files for reference, for example: https://github.com/fossology/fossology/blob/1b8f04f35b11cb3e4d377f8448eaa762abb40473/utils/automation/.gitlab-ci.sample.yml#L1-L5

Comment (imishchuk-carbon) [reply utils/automation/.github-workflow.yml:7]: fixed

File: utils/automation/.github-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# SPDX-FileCopyrightText: © 2020 Siemens AG
+# SPDX-FileCopyrightText: © anupam.ghosh@siemens.com
+# SPDX-FileCopyrightText: © gaurav.mishra@siemens.com
```
Comment (GMishx) [inline utils/automation/.github-workflow.yml:3]:
```suggestion
# SPDX-FileCopyrightText: © 2023 igor.mishchuk@carbonhealth.com
```

**Pull request merged**